File: C++/25_array_subset_of_another.cpp
/*
Idea:
store frequency of each elements for a1 & a2, if any element has more frequency in a2, than a1,
or any new element is found - return false
*/
#include <bits/stdc++.h>
using namespace std;

string isSubset(int a1[], int a2[], int n, int m)
{
    map <int, int>m1, m2;
    //Iterating in both arrays, and storing frequency of each element
    for (int i = 0;i < n;++i)
        m1[a1[i]]++;
    for (int i = 0;i < m;++i)
        m2[a2[i]]++;

    for (auto it : m2)
    {
        auto temp = m1.find(it.first);
        //checking if element is present in first map & it's frequency is under first occurence
        if (temp == m1.end() or it.second > temp->second)
            return "No";
    }
    return "Yes";
}

int main() {
    int t;
    cin >> t;

    while (t--) {
        int n, m;
        cin >> n >> m;
        int a1[n], a2[m];

        for (int i = 0; i < n; i++) {
            cin >> a1[i];
        }
        for (int i = 0; i < m; i++) {
            cin >> a2[i];
        }

        cout << isSubset(a1, a2, n, m) << endl;
    }
    return 0;
}

File: Python/25_array_subset_of_another.py
"""
Idea:
Iterating element in a2, & searching if it's present in a1, if yes, remove it
,if any element in a2, not found in a1. return False
"""


def isSubset(a1, a2, n, m):
    for i in range(m):
        if a2[i] in a1:
            a1.remove(a2[i])
        else:
            return "No"

    return "Yes"


def main():
    T = int(input())

    while T > 0:
        sz = [int(x) for x in input().strip().split()]
        n, m = sz[0], sz[1]
        a1 = [int(x) for x in input().strip().split()]
        a2 = [int(x) for x in input().strip().split()]

        print(isSubset(a1, a2, n, m))

        T -= 1


if __name__ == "__main__":
    main()

# } Driver Code Ends
